inference/worker/utils.py

```diff
@@ -74,28 +87,22 @@ def truncate_prompt(
     with shared_tokenizer_lock:
         ids = tokenizer.encode(prompt)
 
-    max_input_length = worker_config.model_config.max_input_length
-
-    # make room for prompter prefix
-    if plugin_used:
-        max_input_length = max_input_length - 1
+    max_input_length = get_max_input_length(worker_config, plugin_used)
 
     max_total_tokens = worker_config.model_config.max_total_length
     if len(ids) > max_input_length:
-        logger.warning(f"Prompt too long, left-truncating to {max_input_length} tokens")
+        logger.debug(f"Prompt too long, left-truncating to {max_input_length} tokens")
         ids = ids[-(max_input_length - 1) :]
         with shared_tokenizer_lock:
             prompt = tokenizer.decode(ids)
-            # If there is no prompter prefix, due to truncation, add it back.
-            if V2_PROMPTER_PREFIX not in prompt:
-                prompt = V2_PROMPTER_PREFIX + prompt
+            prompt = add_prefix_to_prompt(prompt)
 
     input_length = len(ids)
```

Comment:
As pointed out by Yannic in #3061, this length will turn out to be calculated wrong if the prompter prefix is added above. We should adjust this

Reply:
I've adjusted it to be right before the function returns.

Reply:
I'm not sure if this quite makes sense as a solution. The issue pointed out by Yannic was that the `input_length` considered when computing `spare` tokens is wrong because it doesn't include the prompter prefix, which is added after the initial tokenization. I think this version retains the same flaw. The easiest solution is probably to just to add 1 to input length if the prompter prefix is added.